pkg/http/core/applications.go

```diff
@@ -720,28 +720,96 @@ func listServerGroups(c *gin.Context, wg *sync.WaitGroup, sgs chan ServerGroup,
 		return
 	}
 
+	instanceMap := makeInstanceMap(pods)
+	serverGroups := &unstructured.UnstructuredList{}
+
 	for _, resource := range resources {
-		results, err := client.ListResource(resource, lo)
+		ul, err := client.ListResource(resource, lo)
 		if err != nil {
 			clouddriver.Log(err)
 			continue
 		}
 
-		for _, result := range results.Items {
-			sg := newServerGroup(result, pods, account)
-			sgs <- sg
+		serverGroups.Items = append(serverGroups.Items, ul.Items...)
+	}
+
+	for _, sg := range serverGroups.Items {
+		sg := newServerGroup(sg, instanceMap, account)
+		sgs <- sg
+	}
+}
+
+// makeInstanceMap returns a map of a server group (replicaSet, daemonSet, statefulSet)
+// to a list of instances (pods) that this server group owns.
+//
+// Specifically, the map key is the UID of the owner.
+func makeInstanceMap(pods *unstructured.UnstructuredList) map[string][]Instance {
+	// Map of server group to instances (pods)
+	instanceMap := map[string][]Instance{}
+	// Loop through each pod.
+	for _, pod := range pods.Items {
+		// Loop through each pod's owner reference.
+		p := kubernetes.NewPod(pod.Object)
+		for _, ownerReference := range p.GetObjectMeta().OwnerReferences {
+			key := string(ownerReference.UID)
+			if key == "" {
+				continue
+			}
+
+			if _, ok := instanceMap[key]; !ok {
+				instanceMap[key] = []Instance{}
+			}
```

Comment:
I don't think you need this, because `[]Instance{}` is the default value. 

I commented these lines out locally and ran the tests successfully.